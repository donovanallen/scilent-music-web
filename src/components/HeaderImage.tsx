import React from 'react';
import { IconType } from 'react-icons';

import NextImage from '@/components/NextImage';

import Logo from '~/svg/Logo_Gray.svg';

type HeaderImageProps = {
  imageUrl?: string;
  alt?: string;
  fallbackIcon?: IconType;
};

const HeaderImage: React.FC<HeaderImageProps> = ({
  imageUrl,
  alt,
  fallbackIcon,
}) => {
  const Icon = fallbackIcon;

  return imageUrl ? (
    <div className='relative aspect-square w-28 rounded-md overflow-hidden bg-neutral-700'>
      <NextImage
        src={imageUrl}
        alt={alt || 'Header image'}
        fill
        className='aspect-square object-cover'
        useSkeleton
        priority
      />
    </div>
  ) : Icon ? (
    <>
      <Icon size={36} className='m-auto h-full text-dark' />
    </>
  ) : (
    <div className='relative aspect-square w-28 rounded-md overflow-hidden bg-neutral-700 flex items-center justify-center'>
      <Logo className='w-1/2' />
    </div>
  );
};

export default HeaderImage;
