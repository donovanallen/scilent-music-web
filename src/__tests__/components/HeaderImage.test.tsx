import { render } from '@testing-library/react';
import React from 'react';

import HeaderImage from '@/components/HeaderImage';

describe('HeaderImage', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<HeaderImage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
