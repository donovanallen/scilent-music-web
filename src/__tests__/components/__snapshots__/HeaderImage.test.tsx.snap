// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderImage renders correctly 1`] = `
<DocumentFragment>
  <div
    class="relative aspect-square w-28 rounded-md overflow-hidden bg-neutral-700 flex items-center justify-center"
  >
    <svg
      class="w-1/2"
    />
  </div>
</DocumentFragment>
`;
